s3.go

```diff
@@ -108,22 +118,28 @@ func (m *S3RegionManagedDownloader) Download(dr *DownloadRecord, localFile *os.F
 			Bucket: aws.String(bucket),
 			Key:    aws.String(fname),
 		},
+		s3manager.WithDownloaderRequestOptions(
+			requestInspectorFunc,
```

Comment:
I wanted to use [`WithGetResponseHeader`](https://github.com/aws/aws-sdk-go/blob/a1cf564001efacc3326b79254f0a7378c54959cf/aws/request/request.go#L156) instead of this custom handler, but I noticed that they don't do enough nil checks in there and, sometimes, `req.HTTPResponse.Header` can be nil and causes a panic. I ended up with the custom implementation above, which works well.